HOApp/ViewModel/POEntryViewModel.cs

```diff
@@ -1,106 +1,129 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Collections.ObjectModel;
-using System.Data.Entity;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
+using System.ComponentModel;
 using System.Windows;
 using DataAccess;
-using DataAccess.Entity.Entities;
-using DataAccess.Help;
 using HOApp.Model;
+using System.Data.Entity;
 
 namespace HOApp.ViewModel
 {
-    class POEntryViewModel : DataVMBase
+    internal class POEntryViewModel : DataVMBase
     {
-        public RetailDbContext db = new RetailDbContext();
+        public POEntryViewModel() : base()
+        {
+            _selectableProducts = new List<ProductVM>();
+            PO = new POVM();
+            PropertyChanged += HandlePropertyChanged;
+        }
+
+        private void HandlePropertyChanged(object sender, PropertyChangedEventArgs e)
+        {
+            switch (e.PropertyName)
+            {
+                case nameof(SelectedSupplier):
+                    {
+                        var suppliersId = _selectedSupplier?.TheEntity.SupplierID ?? -1;
+
+                        using (var db = new RetailDbContext())
+                        {
+                            _selectableProducts = db
+                                .Products
+                                .Where(product => product.SupplierID == suppliersId)
+                                .OrderBy(n => n.Description)
+                                .Select(n => new ProductVM { IsNew = false, TheEntity = n })
+                                .ToList();
+                        }
+
+                        RaisePropertyChanged(nameof(SelectableProducts));
+                        PO.SetSupplier(SelectedSupplier);
+                        RaisePropertyChanged(nameof(PO));
+                        break;
+                    }
+                case nameof(SelectedStore):
+                    {
+                        PO.SetStore(SelectedStore);
+                        RaisePropertyChanged(nameof(PO));
+                        break;
+                    }
+            }

```

Comment:
`nameof(SelectedStore)` used instead of "SelectedStore" as it provides some type saftey